### ISSUE TITLE
Wrong yandex_dns_zone resource definition

### ISSUE THREAD

**Issue by evsasha** [yandex-cloud/terraform-provider-yandex#268]
Yandex terraform provide version: 0.75.0
Terraform version: 1.2.1

При создании ресурса типа `yandex_dns_zone` с атрибутом `public = true` зона будет создана сразу с двумя "зонами видимости" - внутренней и внешней.

Тестовый пример:
```hcl
resource "yandex_dns_zone" "external_zone" {
  name        = "externalzone"
  description = "externalzone"
  zone        = "externalzone.site."
  public      = true
  private_networks = []
}

resource "yandex_dns_zone" "internal_zone" {
  name        = "internalzone"
  description = "internalzone"
  zone        = "internalzone.site."
  public      = false
  private_networks = []
}
```
Результат: внешняя зона одновременно и внутренняя! Через Web-UI создавая ресурс руками добиться такого состояния нельзя.
![изображение](https://user-images.githubusercontent.com/23517673/173818552-8ef6becf-6c13-454f-bcb5-8ea52fb70c46.png)

**Comment by thergbway** [comment]
Hello!
This is not a bug.
Terraform always creates `yandex_dns_zone` resources with `public+private` visibility. `Public+private` visibility is needed for special scenarios terraform users may face.
Actually `public + private with 0 private zones` visibility simply equals to `public` visibility.

**Comment by evsasha** [comment]
@thergbway 
But you can't just import a public area created through a web interface and manage it through a terraform. You cannot change the visibility of the zone after it was create! 
Therefore private+public != public.

`Terraform always creates yandex_dns_zone resources with public+private visibility. `
Ok, that may be default behavior but not mandatory.

**Comment by bandikoot** [comment]
+1

After **importing** a dns zone to terraform I've got `.tfstate` like this:
```
{
  "mode": "managed",
  "type": "yandex_dns_zone",
  "name": "zone-name",
  "provider": "provider[\"registry.terraform.io/yandex-cloud/yandex\"]",
  "instances": [
    {
      "schema_version": 0,
      "attributes": {
        "created_at": "2022-09-02T05:15:09Z",
        "description": "oh, this is a very pretty dns zone that has been once created manually via web console by a hasty engineer...",
        "folder_id": "<folder_id>",
        "id": "<zone_id>",
        "labels": {},
        "name": "zone-name",
        "private_networks": [],
        "public": true,
        "timeouts": {
          "create": null,
          "delete": null,
          "update": null
        },
        "zone": "zone-name.ru."
      },
      "sensitive_attributes": [],
      "private": "<private>"
    }
  ]
}
```

The `.tf` file contents:
```
resource "yandex_dns_zone" "zone-name" {
  name        = "zone-name"
  description = "stuff"
  zone        = "zone-name.ru."
  public      = true
}

resource "yandex_dns_recordset" "...
```

However, when there're changes to be applied (e.g., to the "description" field), `terraform apply` fails to perform any action and responds with: `Error: Error while requesting API to update DnsZone "<zone_id>": server-request-id = 5224ff44-a527-4c33-8f11-ad4c60a5553e server-trace-id = 492316238873d1e6:b2ed95ebfe2b3580:492316238873d1e6:1 client-request-id = b1cb206d-3827-483f-b9fa-27a25bf40144 client-trace-id = 5baf3cd7-f61f-49dc-8127-ebb269d89dcd rpc error: code = FailedPrecondition desc = Zone visibility cannot be changed`.

Applying the same changes ("description" field) to a zone that has been originally created via terraform, works fine.

Also, `terraform apply` **stops showing any changes** after the description has been manually updated via web console.

Frankly, it's not something that can be called "correct behavior".

**Comment by stBlackD** [comment]
+1 Невозможно управлять импортированными публичными зонами
Yandex terraform provide version: 0.81.0
Terraform version: 1.2.1

**Comment by ikrivosheev** [comment]
+1, немного не понимаю, что происходит: 

1. в **UI** создана зона: **Публичная**
2. Пытаюсь создать **yandex_compute_instance** c **dns_record**:

```
network_interface {
    nat       = true
    subnet_id = yandex_vpc_subnet.subnet-a.id

    dns_record {
      fqdn = "vm1.example.com."
      dns_zone_id = var.dns_zone_id
      ttl = 100
      ptr = true
    }
  }
```

Получаю: **DNS Zone (...) has no private_visibility**

**Comment by ep4sh** [comment]
+1 Can not manage imported public zones
Terraform v1.4.6
+ provider registry.terraform.io/yandex-cloud/yandex v0.90.0

**Comment by evsasha** [comment]
Fixed in release [v0.102.0](https://github.com/yandex-cloud/terraform-provider-yandex/releases/tag/v0.102.0)
